integration_tests/helpers/constrainedMiningNodeProcess.js

```diff
@@ -0,0 +1,101 @@
+const {getFreePort} = require("./util");
+const dateFormat = require('dateformat');
+const fs = require('fs');
+const {spawnSync, spawn, execSync} = require('child_process');
+const {expect} = require('chai');
+const {createEnv} = require("./config");
+
+let outputProcess;
+
+class ConstrainedMiningNodeProcess {
+    constructor(name, baseNodeAddress, walletAddress ) {
+        this.name = name;
+        this.maxBlocks = 1;
+        this.minDiff = 10000;
+        this.maxDiff = 100000;
+        this.nodeAddress = baseNodeAddress.split(":")[0];
+        this.nodeGrpcPort = baseNodeAddress.split(":")[1];
+        this.walletAddress = walletAddress.split(":")[0];
+        this.walletGrpcPort = walletAddress.split(":")[1];
+    }
+
+    async init(maxBlocks, minDiff, maxDiff) {
+        this.maxBlocks = maxBlocks;
+        this.minDiff = minDiff;
+        this.maxDiff = maxDiff;
+        this.baseDir = `./temp/base_nodes/${dateFormat(new Date(), "yyyymmddHHMM")}/${this.name}`;
+        this.port = await getFreePort(19000, 25000);
+    }
+
+    run(cmd, args, saveFile) {
+        return new Promise((resolve, reject) => {
+            if (!fs.existsSync(this.baseDir)) {
+                fs.mkdirSync(this.baseDir, {recursive: true});
+                fs.mkdirSync(this.baseDir + "/log", {recursive: true});
+            }
+
+            let envs = createEnv(this.name, false, "nodeid.json", this.walletAddress, this.walletGrpcPort, this.port, this.nodeAddress, this.nodeGrpcPort, "8081","127.0.0.1:8084", [], []);
+
+            var ps = spawn(cmd, args, {
+                cwd: this.baseDir,
+                // shell: true,
+                env: {...process.env, ...envs}
+            });
+
+            ps.stdout.on('data', (data) => {
+                //console.log(`stdout: ${data}`);
+                fs.appendFileSync(`${this.baseDir}/log/stdout.log`, data.toString());
+                resolve(ps);
+            });
+
+            ps.stderr.on('data', (data) => {
+                console.error(`stderr: ${data}`);
+                fs.appendFileSync(`${this.baseDir}/log/stderr.log`, data.toString());
+            });
+
+            ps.on('close', (code) => {
+                let ps = this.ps;
+                this.ps = null;
+                if (code) {
+                    console.log(`child process exited with code ${code}`);
+                    reject(`child process exited with code ${code}`);
+                } else {
+                    resolve(ps);
+                }
+            });
+
+            expect(ps.error).to.be.an('undefined');
+            this.ps = ps;
+        });
+    }
+
+    async startNew() {
```

Comment:
This method requires that init has been called before it. Perhaps you should merge that code in here, or add a call to init here. 

The other nodes require a separate `init` step and that's why they are separate. This process does not require the additional step, so you can maybe put the min/max args directly in here

Reply:
The step which determines the number of blocks to be mined is decoupled from the creation of the miner, hence the separate step.

Will add defaults, though think this will still be needed in the tests.